ringbuf/reader.go

```diff
@@ -167,16 +169,25 @@ func (r *Reader) Read() (Record, error) {
 	}
 
 	for {
-		_, err := r.poller.Wait(r.epollEvents)
-		if err != nil {
-			return Record{}, err
+		if !r.haveData {
+			_, err := r.poller.Wait(r.epollEvents[:cap(r.epollEvents)])
+			if err != nil {
+				return Record{}, err
+			}
+			r.haveData = true
 		}
 
-		record, err := readRecord(r.ring)
-		if errors.Is(err, errBusy) || errors.Is(err, errDiscard) {
-			continue
+		for {
+			record, err := readRecord(r.ring)
+			if err == errBusy || err == errDiscard {
+				continue
+			}
+			if err == errEOR {
```

Comment:
Is this a left over from previous experiments? `readRecord()` does not return `errEOR` afaict.

Other than that, the changes work fine here.